File: GFG/algorithms/searching/linearSearch.py
#py3 code to do linear search of x in arr[]
#otherwise return 0
"""linear search"""
def linearSearch(arr,n,x):
    for i in range(0,n):
        if(arr[i]==x):
            return i
        return 0
#driver code
arr = [2,3,10,7,8,9]
x = 10
n = len(arr)
m = linearSearch(arr,n,x)
print(m)

File: GFG/algorithms/searching/linearSearch.cpp
// C++ code to linearly search x in arr[]. If x
// is present then return its location, otherwise
// return -1

/*  
Start from the leftmost element of arr[] and one by one compare x with each element of arr[]
If x matches with an element, return the index.
If x doesn’t match with any of the elements, return -1.
*/

#include <iostream>
using namespace std;

int linearSearch(int arr[], int n, int x)
{
	int i;
	for (i = 0; i < n; i++)
		if (arr[i] == x)
			return i;
	return -1;
}

// Driver code
int main(void)
{
	int arr[] = { 2, 3, 4, 10, 40 };
	int x = 10;
	int n = sizeof(arr) / sizeof(arr[0]);

	// Function call
	int result = linearSearch(arr, n, x);
	(result == -1)
		? cout << "Element is not present in array"
		: cout << "Element is present at index " << result;
	return 0;
}
